packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -0,0 +1,411 @@
+                    }
+                    summaryObject.content = "";
+                    // If the cache has the hash of the blob and handle of last summary is also present, then use that
+                    // cached path for the given blob. Also update the caches for future use.
+                    if (cachedPath === undefined || parentHandle === undefined) {
+                        blobs++;
+                    } else {
+                        reusedBlobs++;
+                        id = `${parentHandle}/${cachedPath}`;
+                        value = undefined;
+                    }
+                    blobTreeDedupCachesLatest.blobShaToPath.set(hash, currentPath);
+                    blobTreeDedupCachesLatest.pathToBlobSha.set(currentPath, hash);
+                    break;
+                }
+                case api.SummaryType.Handle: {
+                    if (!parentHandle) {
+                        throw Error("Parent summary does not exist to reference by handle.");
+                    }
+                    let handlePath = summaryObject.handle;
+                    if (handlePath.length > 0 && !handlePath.startsWith("/")) {
+                        handlePath = `/${handlePath}`;
+                    }
+                    const pathKey = `${rootNodeName}${handlePath}`;
+                    // We try to get the summary tree from the cache so that we can expand it in order to dedup the blobs.
+                    // We always send whole tree no matter what, even if some part of the tree did not change in order to dedup
+                    // the blobs.
+                    const summaryTreeToExpand = this.blobTreeDedupCaches.treesPathToTree.get(pathKey);
+                    if (summaryTreeToExpand !== undefined && allowHandleExpansion) {
+                        blobTreeDedupCachesLatest.treesPathToTree.set(currentPath, summaryTreeToExpand);
+                        const result = await this.convertSummaryToSnapshotTree(
+                            parentHandle,
+                            summaryTreeToExpand,
+                            blobTreeDedupCachesLatest,
+                            rootNodeName,
+                            allowHandleExpansion,
+                            currentPath,
+                            true);
+                        value = result.snapshotTree;
+                        reusedBlobs += result.reusedBlobs;
+                        blobs += result.blobs;
+                    } else {
+                        // Ideally we should not come here as we should have found it in cache. But in order to successfully upload the summary
+                        // we are just logging the event. Once we make sure that we don't have any telemetry for this, we would remove this.
+                        this.logger.sendTelemetryEvent({ eventName: "SummaryTreeHandleCacheMiss", parentHandle, handlePath: pathKey });
```

Comment:
Should this be an error event? I.e. if this is unexpected and points to bug in the logic, then it absolutely need to be an error event